src/carplay.cs

```diff
@@ -0,0 +1,719 @@
+
+		[Export ("mapTemplate:selectedPreviewForTrip:usingRouteChoice:")]
+		void SelectedPreview (CPMapTemplate mapTemplate, CPTrip trip, CPRouteChoice routeChoice);
+
+		[Export ("mapTemplate:startedTrip:usingRouteChoice:")]
+		void StartedTrip (CPMapTemplate mapTemplate, CPTrip trip, CPRouteChoice routeChoice);
+
+		[Export ("mapTemplateDidCancelNavigation:")]
+		void DidCancelNavigation (CPMapTemplate mapTemplate);
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CPNavigationAlert : NSSecureCoding {
+
+		// Defined inside CPNavigationAlert.h
+		// static NSTimeInterval const CPNavigationAlertMinimumDuration = 10;
+		[Static]
+		[Wrap ("10D")]
+		double NavigationAlertMinimumDuration { get; }
+
+		[Export ("initWithTitleVariants:subtitleVariants:image:priority:primaryAction:secondaryAction:duration:")]
+		IntPtr Constructor (string [] titleVariants, [NullAllowed] string [] subtitleVariants, [NullAllowed] UIImage image, CPNavigationAlertPriority priority, CPAlertAction primaryAction, [NullAllowed] CPAlertAction secondaryAction, double duration);
+
+		[Export ("alertPriority")]
+		CPNavigationAlertPriority AlertPriority { get; }
+
+		[Export ("titleVariants", ArgumentSemantic.Copy)]
+		string [] TitleVariants { get; }
+
+		[Export ("subtitleVariants", ArgumentSemantic.Copy)]
+		string [] SubtitleVariants { get; }
+
+		[NullAllowed, Export ("image", ArgumentSemantic.Copy)]
+		UIImage Image { get; }
+
+		[Export ("primaryAction", ArgumentSemantic.Strong)]
+		CPAlertAction PrimaryAction { get; }
+
+		[NullAllowed, Export ("secondaryAction", ArgumentSemantic.Strong)]
+		CPAlertAction SecondaryAction { get; }
+
+		[Export ("duration")]
+		double Duration { get; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CPTravelEstimates : NSSecureCoding {
+
+		[Export ("initWithDistanceRemaining:timeRemaining:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSMeasurement<NSUnitLength> distance, double time);
+
+		[Export ("distanceRemaining", ArgumentSemantic.Copy)]
+		NSMeasurement<NSUnitLength> DistanceRemaining { get; }
+
+		[Export ("timeRemaining")]
+		double TimeRemaining { get; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CPNavigationSession {
+
+		[Export ("pauseTripForReason:")]
+		void PauseTrip (CPTripPauseReason reason);
+
+		[Export ("finishTrip")]
+		void FinishTrip ();
+
+		[Export ("cancelTrip")]
+		void CancelTrip ();
+
+		[Export ("upcomingManeuvers", ArgumentSemantic.Copy)]
+		CPManeuver [] UpcomingManeuvers { get; set; }
+
+		[Export ("trip", ArgumentSemantic.Strong)]
+		CPTrip Trip { get; }
+
+		[Export ("updateTravelEstimates:forManeuver:")]
+		void UpdateTravelEstimates (CPTravelEstimates estimates, CPManeuver maneuver);
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (CPTemplate))]
+	interface CPSearchTemplate {
```

Comment:
same, base ctor ? disable default ctor ?

Reply:
Same, I do not think there is other base ctor other than init here

```csharp
[Abstract]
	[NoWatch, NoTV, NoMac, iOS (12,0)]
	[BaseType (typeof (NSObject))]
	interface CPTemplate : NSSecureCoding {

	}
```